components_library/processors/complex_generator_core.py

```diff
@@ -0,0 +1,116 @@
+from m5.util.convert import toLatency, toMemoryBandwidth
+
+from components_library.processors.cpu_types import CPUTypes
+from components_library.processors.py_generator_core import PyGeneratorCore
+
+
+from .py_generator_core import PyGeneratorCore
+
+
+class ComplexGeneratorCore(PyGeneratorCore):
+    def __init__(
+        self,
+    ):
+        super(ComplexGeneratorCore, self).__init__(CPUTypes.CMXGEN)
+        self._main_traffic = []
+
+    def add_linear(
+        self,
+        duration="1ms",
+        rate="100GB/s",
+        block_size=64,
+        min_addr=0,
+        max_addr=32768,
+        rd_perc=100,
+        data_limit=0,
+    ):
```

Comment:
Let's talk about what is good to have a "reasonable default" vs required. Some of these you can't guess for the user.